cmake/scripts/linux/ArchSetup.cmake

```diff
@@ -88,7 +96,11 @@ if(CMAKE_BUILD_TYPE STREQUAL Release OR CMAKE_BUILD_TYPE STREQUAL MinSizeRel)
 
           set(CMAKE_CXX_COMPILE_OPTIONS_IPO -flto=thin)
           set(CMAKE_C_COMPILE_OPTIONS_IPO -flto=thin)
-          set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--thinlto-jobs=${NJOBS},--thinlto-cache-dir=${CLANG_LTO_CACHE}")
+          if(LLD_FOUND)
+            set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--thinlto-jobs=${NJOBS},--thinlto-cache-dir=${CLANG_LTO_CACHE}")
```

Comment:
I think you can pass the same options to lld as mold, because lld accepts both `--plugin-opt=` style options and the ones without the `--plugin-opt=` prefix.